src/js/components/Diagram/index.d.ts

```diff
@@ -6,8 +6,32 @@ export type DiagramConnectionAnchor = 'center' | 'vertical' | 'horizontal';
 export type DiagramConnectionType = 'direct' | 'curved' | 'rectilinear';
 
 export interface DiagramProps {
+  animation?:
+    | boolean
+    | 'pulse'
+    | 'draw'
+    | {
+        type?: 'pulse' | 'draw';
+        // looking into it, but delay seems to only work when delay <= 1000
+        delay?: number;
+        duration?: number;
+        // currently, the only animation Diagram has that utilizes size is for "pulse"
+        size?: 'xsmall' | 'small' | 'medium' | 'large' | 'xlarge';
+      };
   connections: {
     anchor?: DiagramConnectionAnchor;
+    animation?:
+      | boolean
+      | 'pulse'
+      | 'draw'
+      | {
+          type?: 'pulse' | 'draw';
+          // looking into it, but delay seems to only work when delay <= 1000
+          delay?: number;
+          duration?: number;
+          // currently, the only animation Diagram has that utilizes size is for "pulse"
+          size?: 'xsmall' | 'small' | 'medium' | 'large' | 'xlarge';
+        };
```

Comment:
Shouldn't be duplicated

Reply:
Now consolidated into a variable called `DiagramAnimationType` that gets plugged into both the component level, and connection level